src/payment/strategies/bolt/bolt-payment-strategy.ts

```diff
@@ -267,6 +287,16 @@ export default class BoltPaymentStrategy implements PaymentStrategy {
         return boltEmbedded;
     }
 
+    private _hasBoltAccount() {
+        const state = this._store.getState();
+        const customer = state.customer.getCustomer();
+        const billingAddress = state.billingAddress.getBillingAddress();
+        const email = customer?.email || billingAddress?.email || '';
+        const boltClient = this._getBoltClient();
+
+        return boltClient.hasBoltAccount(email);
```

Comment:
Do we need to do some kind of check here, if this call fails?

Reply:
Sure. I think that we need to add some error handling here